tests/basic/condvar.rs

```diff
@@ -32,6 +32,30 @@ fn notify_one() {
     )
 }
 
+#[test]
+fn notify_one_while() {
+    check_dfs(
+        || {
+            let lock = Arc::new(Mutex::new(false));
+            let cond = Arc::new(Condvar::new());
+
+            {
+                let lock = Arc::clone(&lock);
+                let cond = Arc::clone(&cond);
+                thread::spawn(move || {
+                    let guard = lock.lock().unwrap();
+                    cond.wait_while(guard, |flag| !*flag).unwrap();
+                });
```

Comment:
Maybe add an assertion here to check that the boolean is now true. (Also in the test below.)

Reply:
Yes, good idea, done.